etna/core/blocks/cta.py

```diff
@@ -29,3 +32,38 @@ def get_context(self, value, parent_context=None):
             link_pages.append(page_2.specific)
         context["link_pages"] = link_pages
         return context
+
+
+class ButtonBlock(blocks.StructBlock):
+    label = blocks.CharBlock()
+    link = APIPageChooserBlock(required=False)
+    external_link = blocks.URLBlock(required=False)
```

Comment:
Would it be possible to get just one field back in the API?

I think these might be better as `page_link`/`internal_link` and `external_link` but returned in the API as just `href` (or something) so I don't have to make conditionals.

What do you think?

Reply:
Good idea, I was trying to think of a way to simplify it but couldn't think of anything good. Will add that now!

Reply:
![image](https://github.com/nationalarchives/ds-wagtail/assets/62654785/78b6ef47-6657-4847-8bf8-30739717d40a)

How's this?

Reply:
Perfect! 🤩